sdk/tables/azure-tables/review/tables.api.md

```diff
@@ -4,8 +4,595 @@
+    // (undocumented)
+    xMsRequestId?: string;
+    // (undocumented)
+    xMsVersion?: string;
+}
+
+// @public
+export interface TableQueryEntitiesOptionalParams extends coreHttp.OperationOptions {
+    nextPartitionKey?: string;
+    nextRowKey?: string;
+    queryOptions?: QueryOptions;
+    requestId?: string;
+    timeout?: number;
+}
+
+// @public
+export type TableQueryEntitiesResponse = TableQueryEntitiesHeaders & TableEntityQueryResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: TableEntityQueryResponse;
+        parsedHeaders: TableQueryEntitiesHeaders;
+    };
+};
+
+// @public
+export interface TableQueryEntitiesWithPartitionAndRowKeyHeaders {
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    eTag?: string;
+    // (undocumented)
+    xMsClientRequestId?: string;
+    // (undocumented)
+    xMsContinuationNextPartitionKey?: string;
+    // (undocumented)
+    xMsContinuationNextRowKey?: string;
+    // (undocumented)
+    xMsRequestId?: string;
+    // (undocumented)
+    xMsVersion?: string;
+}
+
+// @public
+export interface TableQueryEntitiesWithPartitionAndRowKeyOptionalParams extends coreHttp.OperationOptions {
+    queryOptions?: QueryOptions;
+    requestId?: string;
+    timeout?: number;
+}
+
+// @public
+export type TableQueryEntitiesWithPartitionAndRowKeyResponse = TableQueryEntitiesWithPartitionAndRowKeyHeaders & TableEntityQueryResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: TableEntityQueryResponse;
+        parsedHeaders: TableQueryEntitiesWithPartitionAndRowKeyHeaders;
+    };
+};
+
+// @public
+export interface TableQueryHeaders {
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    xMsClientRequestId?: string;
+    // (undocumented)
+    xMsContinuationNextTableName?: string;
+    // (undocumented)
+    xMsRequestId?: string;
+    // (undocumented)
+    xMsVersion?: string;
+}
+
+// @public
+export type TableQueryOperationResponse = TableQueryHeaders & TableQueryResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: TableQueryResponse;
+        parsedHeaders: TableQueryHeaders;
+    };
+};
+
+// @public
+export interface TableQueryOptionalParams extends coreHttp.OperationOptions {
+    nextTableName?: string;
+    queryOptions?: QueryOptions;
+    requestId?: string;
+}
+
+// @public
+export interface TableQueryResponse {
+    odataMetadata?: string;
+    value?: TableResponseProperties[];
+}
+
+// @public
+export type TableResponse = TableResponseProperties & {
+    odataMetadata?: string;
+};
+
+// @public
+export interface TableResponseProperties {
+    odataEditLink?: string;
+    odataId?: string;
+    odataType?: string;
+    tableName?: string;
+}
+
+// @public (undocumented)
+export class TableServiceClient {
+    constructor(url: string, options?: GeneratedClientOptionalParams | undefined);
```

Comment:
_(I get that this is still very much a work in progress, so don't feel the need to react to this right away, unless others call it out as something that we should address eagerly..)_

I think we need better names for these option types.  In general, it feels like our pattern is to use `Options` and not `OptionalParams` as the suffix.  I would also expect the prefix to be tied to the operation.  So for this type, I would name this type `TableServiceClientOptions`.

For the individual operations, I would use names like `CreateEntityOptions` or `GetAccessPolicyOptions`, I think this will be more in line with other libraries.

Reply:
I think it would be a good idea to rename these now. @mahdiva you should be able to do so by using import aliases

```typescript
import { GeneratedClientOptionalParams as TableServiceClientOptions } from "./generated/models"
```